### PR TITLE
Table requests calls

### PR DESCRIPTION
**Pull request by pthegner**
runtime w/o sql : 16 minutes 55 seconds (1015 seconds) (Window with docker)

### CODE REVIEW

**Comment by vincentlepot** [comment]
I'm quite sure this can be faster with SparkSQL (that's what we experience with the other tables)

**Comment by dvdn** [comment]
(intel core i5) w/o sql : 366 seconds

**Comment by pthegner** [comment]
with sql : 531 seconds -> 8 minutes 51 seconds

File: requests_calls_sql.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    when(df.end_point_id.isNull() , 1).when(df.end_point_id == 0, 1).otherwise(df.end_point_id).alias('end_point_id'),
+    when(df.journeys.isNull(), 1).otherwise(0).alias('n_w_j'),
+    when(df.info_response.isNull(), 0).otherwise(when(df.info_response.object_count.isNull(), 0).otherwise(df.info_response.object_count)).alias('o_c'),
+).withColumn("nb", lit(1))\
+    .groupBy("region_id", "api", "user_id", "application_name", "is_internal_call", "request_date", "end_point_id").agg({"nb": "sum", "n_w_j": "sum", "o_c": "sum"})
+
+#DB insert
+conn = psycopg2.connect(common.get_db_connection_string())
+cur = conn.cursor()
+cur.execute("DELETE FROM stat_compiled.requests_calls WHERE request_date >= %s AND request_date <= %s", (treatment_day_start, treatment_day_end))
```
Comment (nberard) [inline requests_calls_sql.py]: you should use common functions to handle rollback

File: requests_calls_sql.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+requests_calls = df.select(
+    when(df.coverages[0].region_id.isNull(), '').otherwise(df.coverages[0].region_id).alias('region_id'),
+    df.api,
+    df.user_id,
+    df.application_name,
+    when(df.user_name == 'canaltp', 1).otherwise(0).alias('is_internal_call'),
+    from_unixtime(df.request_date, 'yyyy-MM-dd').alias('request_date'),
+    when(df.end_point_id.isNull() , 1).when(df.end_point_id == 0, 1).otherwise(df.end_point_id).alias('end_point_id'),
+    when(df.journeys.isNull(), 1).otherwise(0).alias('n_w_j'),
+    when(df.info_response.isNull(), 0).otherwise(when(df.info_response.object_count.isNull(), 0).otherwise(df.info_response.object_count)).alias('o_c'),
```
Comment (nberard) [inline requests_calls_sql.py]: o_c ?

File: requests_calls_sql.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+).withColumn("nb", lit(1))\
+    .groupBy("region_id", "api", "user_id", "application_name", "is_internal_call", "request_date", "end_point_id").agg({"nb": "sum", "n_w_j": "sum", "o_c": "sum"})
```
Comment (nberard) [inline requests_calls_sql.py]: n_w_j ?

File: requests_calls_sql.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+#DB insert
+conn = psycopg2.connect(common.get_db_connection_string())
+cur = conn.cursor()
+cur.execute("DELETE FROM stat_compiled.requests_calls WHERE request_date >= %s AND request_date <= %s", (treatment_day_start, treatment_day_end))
+cur.close()
+insert_cur = conn.cursor()
+
+sql_query_fmt = """
```
Comment (nberard) [inline requests_calls_sql.py]: same, common insert function can be used

File: requests_calls_sql.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import sys, os
+sys.path.append(os.path.abspath("includes"))
+from time import time
+from includes import common
+import psycopg2
```
Comment (nberard) [inline requests_calls_sql.py]: can be removed

**Pull request merged**